apps/server/src/modules/group/uc/group.uc.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from '@nestjs/common';
+import { EntityId, Page, Permission, SchoolDO, SortOrder, User, UserDO } from '@shared/domain';
+import { AuthorizationContextBuilder, AuthorizationService } from '@src/modules/authorization';
+import { ClassService } from '@src/modules/class';
+import { Class } from '@src/modules/class/domain';
+import { RoleService } from '@src/modules/role';
+import { RoleDto } from '@src/modules/role/service/dto/role.dto';
+import { SchoolService } from '@src/modules/school';
+import { SystemDto, SystemService } from '@src/modules/system';
+import { UserService } from '@src/modules/user';
+import { Group, GroupUser } from '../domain';
+import { GroupService } from '../service';
+import { SortHelper } from '../util';
+import { ClassInfoDto, ResolvedGroupUser } from './dto';
+import { GroupUcMapper } from './mapper/group-uc.mapper';
+
+@Injectable()
+export class GroupUc {
+	constructor(
+		private readonly groupService: GroupService,
+		private readonly classService: ClassService,
+		private readonly systemService: SystemService,
+		private readonly userService: UserService,
+		private readonly roleService: RoleService,
+		private readonly schoolService: SchoolService,
+		private readonly authorizationService: AuthorizationService
+	) {}
+
+	public async findClassesForSchool(
+		userId: EntityId,
+		schoolId: EntityId,
+		skip = 0,
+		limit?: number,
+		sortBy: keyof ClassInfoDto = 'name',
+		sortOrder: SortOrder = SortOrder.asc
+	): Promise<Page<ClassInfoDto>> {
+		const school: SchoolDO = await this.schoolService.getSchoolById(schoolId);
+
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkPermission(user, school, AuthorizationContextBuilder.read([Permission.CLASS_LIST]));
+
+		let combinedClassInfo: ClassInfoDto[] = await this.findCombinedClassListForSchool(schoolId);
+
+		combinedClassInfo = combinedClassInfo.sort((a: ClassInfoDto, b: ClassInfoDto): number =>
+			SortHelper.genericSortFunction(a[sortBy], b[sortBy], sortOrder)
+		);
+
+		const pageContent: ClassInfoDto[] = combinedClassInfo.slice(skip, limit ? skip + limit : combinedClassInfo.length);
+
+		const page: Page<ClassInfoDto> = new Page<ClassInfoDto>(pageContent, combinedClassInfo.length);
+
+		return page;
+	}
+
+	private async findCombinedClassListForSchool(schoolId: string): Promise<ClassInfoDto[]> {
+		const classes: Class[] = await this.classService.findClassesForSchool(schoolId);
+		const groupsOfTypeClass: Group[] = await this.groupService.findClassesForSchool(schoolId);
+
+		const systemMap: Map<EntityId, SystemDto> = await this.findSystemNamesForGroups(groupsOfTypeClass);
+
+		const classInfosFromGroups: ClassInfoDto[] = await Promise.all(
```

Comment:
I would move classInfosFromGroups and classInfosFromClasses to one or two seperate functions. Then this would be better readable